### PR TITLE
Test against the latest release of ruby, 2.4.0.

### PR DESCRIPTION
**Pull request by matthewshafer**
Update the other versions of ruby to test against the latest patch version.
Update .ruby_version to use the latest version of ruby.

@yarmand

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 rvm:
   - 1.9.3
   - 2.0.0
-  - 2.1
-  - 2.2
-  - 2.2.2
+  - 2.1.10
+  - 2.2.6
+  - 2.3.3
+  - 2.4.0
```
Comment (yarmand) [inline .travis.yml]: isn't travis building against latest patch version when you do not specify them ?
By specifying ourselves, we need to update all our repo when a new patch is released.

Comment (yarmand) [inline .travis.yml]: well ... travis is not using latest patches :(
I did a test on one of my personal repo, specifying major.minor version and print 'ruby -v' as pre-script.
It gives me  ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-linux] for ruby 2.1.

Comment (sideshowcoder) [inline .travis.yml]: Maybe consider including `ruby-head`? see http://rubies.travis-ci.org/

Comment (matthewshafer) [inline .travis.yml]: @sideshowcoder that works for me, I tend to put ruby-head builds to allow failure if that's okay?

Comment (sideshowcoder) [inline .travis.yml]: Makes sense to me @matthewshafer 👍

**Comment by matthewshafer** [comment]
added testing against ruby head with allowing failures.  Didn't turn on fast-fail because the test suite is pretty quick already.

**Pull request merged**